sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala

```diff
@@ -290,6 +291,7 @@ case class DropViewStatement(
 case class InsertIntoStatement(
     table: LogicalPlan,
     partitionSpec: Map[String, Option[String]],
+    userSpecifiedCols: Seq[Attribute],
```

Comment:
I'm not sure if this should be `Attribute` not `String`. `InsertIntoStatement`'s child is `query`, not `table`, which means these attributes will be resolved based on `query`, while they should be resolved by `table`.

Reply:
ah, `String` here is much simpler and more rational.